pkg/helm/helm.go

```diff
@@ -7,125 +7,119 @@ import (
 	minNumDescriptors++
+	allowedMediaTypes = append(allowedMediaTypes, helmregistry.ChartLayerMediaType, helmregistry.LegacyChartLayerMediaType)
 
-	allowedMediaTypes = append(allowedMediaTypes, ChartLayerMediaType, LegacyChartLayerMediaType)
-	manifest, descriptors, err := oras.Pull(ctx(os.Stdout, log.GetLevel() == log.TraceLevel), p.Resolver, ociFullName, store,
+	var descriptors, layers []ocispec.Descriptor
+	registryStore := content.Registry{Resolver: p.Resolver}
+
+	context := orascontext.WithLoggerFromWriter(context.Background(), os.Stdout)
+	manifest, err := oras.Copy(context, registryStore, parsedRef.String(), memoryStore, "",
 		oras.WithPullEmptyNameAllowed(),
-		oras.WithAllowedMediaTypes(allowedMediaTypes))
+		oras.WithAllowedMediaTypes(allowedMediaTypes),
+		oras.WithLayerDescriptors(func(l []ocispec.Descriptor) {
+			layers = l
+		}))
 	if err != nil {
 		return nil, "", err
 	}
+
+	descriptors = append(descriptors, manifest)
+	descriptors = append(descriptors, layers...)
+
 	numDescriptors := len(descriptors)
 	if numDescriptors < minNumDescriptors {
-		return nil, manifest.Digest.String(), errors.New(
-			fmt.Sprintf("manifest does not contain minimum number of descriptors (%d), descriptors found: %d",
-				minNumDescriptors, numDescriptors))
+		return nil, "", fmt.Errorf("manifest does not contain minimum number of descriptors (%d), descriptors found: %d",
+			minNumDescriptors, numDescriptors)
 	}
+	var configDescriptor *ocispec.Descriptor
 	var chartDescriptor *ocispec.Descriptor
 	for _, descriptor := range descriptors {
 		d := descriptor
 		switch d.MediaType {
-		case ChartLayerMediaType:
+		case helmregistry.ConfigMediaType:
+			configDescriptor = &d
+		case helmregistry.ChartLayerMediaType:
 			chartDescriptor = &d
-		case LegacyChartLayerMediaType:
+		case helmregistry.LegacyChartLayerMediaType:
 			chartDescriptor = &d
-			log.Warnf("Warning: chart media type %s is deprecated\n", LegacyChartLayerMediaType)
+			log.Infof("Warning: chart media type %s is deprecated\n", helmregistry.LegacyChartLayerMediaType)
 		}
 	}
+	if configDescriptor == nil {
+		return nil, "", fmt.Errorf("could not load config with mediatype %s", helmregistry.ConfigMediaType)
+	}
 	if chartDescriptor == nil {
-		return nil, manifest.Digest.String(), errors.New(
-			fmt.Sprintf("manifest does not contain a layer with mediatype %s or %s",
-				ChartLayerMediaType, LegacyChartLayerMediaType))
+		return nil, "", fmt.Errorf("manifest does not contain a layer with mediatype %s",
+			helmregistry.ChartLayerMediaType)
 	}
-	_, chartData, ok := store.Get(*chartDescriptor)
+
+	_, chartData, ok := memoryStore.Get(*chartDescriptor)
 	if !ok {
 		return nil, manifest.Digest.String(), errors.Errorf("Unable to retrieve blob with digest %s", chartDescriptor.Digest)
 	}
 
 	return bytes.NewBuffer(chartData), manifest.Digest.String(), nil
 }
+
+// Code from Helm Registry Client. Copied here since it belonged to a internal package.
```

Comment:
Here would be an ideal spot to reference an upstream issue discussing this.

Reply:
> t'd be worth creating an issue to begin the conversation upstream and see whether it's something they're keen to do or at least see done.

Done: https://github.com/helm/helm/issues/10623